src/realm/sync/client.cpp

```diff
@@ -429,7 +431,7 @@ void ClientImpl::cancel_reconnect_delay()
             }
         }
     };
-    get_service().post(std::move(handler)); // Throws
+    event_loop_post(std::move(handler)); // Throws
```

Comment:
it's actually ever so slightly more performant to do 
```event_loop_post([this](Status status) {});```
Than it is to do
```
auto handler = [this](Status) {};
event_loop_post(std::move(handler));
```

This is def a nit, but if we're changing all these call sites anyways, might as well wrap them up in the slightly nicer war?